### PR TITLE
[FIX] web: barcode multiple error messages

### PR DESCRIPTION
**Pull request by ande-odoo**
Current behaviour:
---
When trying to scan a barcode, if you open
and close the pop-up window multiple time. (ie 5)
When you'll scan an invalid barcode, you'll get
multiple error messages. (5)

Expected behaviour:
---
Only one error message.

Steps to reproduce:
---
1. Go to Barcode
2. Click on Tap to scan
3. Allow the camera uses
4. Click on the closing cross
5. Repeat a few time (5 by example)
6. Click on Tap to scan
7. Scan an invalid barcode
8. 5 error messages (No picking ...)

Cause of the issue:
---
Promises weren't resolved until a barcode
(valid or invalid) was shown to the camera.
Which made the promises resolve all at once.

Fix:
---
Resolve the promise if the closing cross is clicked

opw-3634480

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by nle-odoo — APPROVED** [review]
looks good to me

it seemed a little unexpected to do:

```py
promise.resolve(result.barcode);
```
with `undefined` when we close the popup, but all the `barcode = await scanBarcode` seems to have a `if(barcode) {} else {}` so good to me

**Review by aab-odoo — APPROVED** [review]
robodoo delegate=ande-odoo

File: addons/web/static/src/webclient/barcode/barcode_scanner.js [hunk 1]
```diff
@@ -205,18 +200,22 @@ export function isBarcodeScannerSupported() {
...
         translateFn: _t,
         props: {
-            onClose: () => appForBarcodeDialog.destroy(),
+            onClose: (result = {}) => {
+                appForBarcodeDialog.destroy();
+                if (result.error) {
+                    promise.reject({ error: result.error });
+                } else {
+                    promise.resolve(result.barcode);
+                }
```
[the remark this answers is not in the document]
Comment (ande-odoo) [reply addons/web/static/src/webclient/barcode/barcode_scanner.js:217]: By doing your suggestion, if result.barcode is undefined, the way we use scanBarcode with an await, the promise would never return and we would never get the notification
https://github.com/odoo/enterprise/blob/508266cb5776206f382aea53ba1c1468ecc79d1f/stock_barcode/static/src/components/main.js#L191

**Comment by ande-odoo** [comment]
@robodoo r+

**Pull request closed without merging**